pkgs/development/compilers/ponyc/default.nix

```diff
@@ -91,7 +92,12 @@ stdenv.mkDerivation (rec {
     # Replace downloads with local copies.
     substituteInPlace lib/CMakeLists.txt \
         --replace-fail "https://github.com/google/benchmark/archive/v$benchmarkRev.tar.gz" "$NIX_BUILD_TOP/deps/benchmark-$benchmarkRev.tar" \
-        --replace-fail "https://github.com/google/googletest/archive/release-$googletestRev.tar.gz" "$NIX_BUILD_TOP/deps/googletest-$googletestRev.tar"
+        --replace-fail "https://github.com/google/googletest/archive/refs/tags/v$googletestRev.tar.gz" "$NIX_BUILD_TOP/deps/googletest-$googletestRev.tar"
+
+    # The build system will run $CMAKE_CXX_COMPILER --print-file-name=libstdc++.a here, which used
+    # to work due to GCC parts leaking into Clang. This has been fixed, so we must now use gcc here.
+    substituteInPlace src/libponyc/CMakeLists.txt \
+        --replace-fail "\''${CMAKE_CXX_COMPILER}" "${gcc}/bin/gcc"
```

Comment:
Can we use `${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc` here?

Reply:
That causes a build failure because this unfortunately only works with GCC (whereas `stdenv` uses Clang here):
```
$ nix run nixpkgs#gcc -- --print-file-name=libstdc++.a
/nix/store/2agih0y5ns3sgbviw2xhivdgg59b41g9-gcc-14-20241116/lib/gcc/x86_64-unknown-linux-gnu/14.2.1/../../../libstdc++.a
$ nix run nixpkgs#clang -- --print-file-name=libstdc++.a
libstdc++.a # Then: file not found
```
That's the reason this substitute was added. It uses Clang without this susbstitute and in a previous nixpkgs version, that would return the same path as GCC does nowadays (which isn't shipped with Clang but with GCC, so that was a "leak" and was fixed).

Reply:
There's a way to get clang without libcxx, I just don't remember how. What can be done is in `all-packages.nix`, you can do an if `stdenv.cc.isClang` and use an stdenv from `llvmPackages`.

Reply:
I don't really understand. Isn't the problem that clang does *not* have libcxx so it just echoes back the file name it doesn't know about? Wouldn't we want clang *with* libcxx?

Reply:
Not quite, we have variants of LLVM stdenvs. Clang isn't ultimately dependent on libstdc++ nor libc++. The wrapper handles it.

Reply:
I don't really know what's causing things to not work correctly without looking at this but I know the previous way wasn't correct.

Reply:
@numinit It would be awesome to get your feedback on this. Do you know of some way to get this working with clang, or should we stick to GCC?

Reply:
I can take a look at this in a couple days. I'll assign myself so I know to come back.

Reply:
Yeah, the stdenv weirdness is more Tristan's wheelhouse, though the rest of the PR looks good and has my approval. Thanks for the fix.

Reply:
Ok, https://github.com/ponylang/ponyc/pull/4679 might fix it. I'm currently testing things.